crates/ra_project_model/src/cargo_workspace.rs

```diff
@@ -171,7 +170,7 @@ impl CargoWorkspace {
         if let Some(parent) = cargo_toml.parent() {
             meta.current_dir(parent);
         }
-        let meta = meta.exec().map_err(|e| format!("cargo metadata failed: {}", e))?;
+        let meta = meta.exec().map_err(|e| anyhow!("cargo metadata failed: {}", e))?;
```

Comment:
I'd suggest to check that we get a good error message here. I am concerned because we got a report #3118 where the error message was not very helpful... I also believe `meta.exec().context("cargo metadata failed")` would give us a bit better error message, though not sure...

Reply:
Agree, I *think* this should use `.context`

Reply:
I ran the function with a path which wasn't pointing to a `Cargo.toml` file and neither setup records he path that was used in the error message. Here's what I got with your change:
```
cargo metadata failed

Caused by:
    Error during execution of `cargo metadata`: error: the manifest-path must be a path to a Cargo.toml file
``` 
Maybe it would make sense to add a new method on `MetadataCommand` which returns the full command called and include that in the error message?

I can't tell from that bug report exactly what info would have clarified things.

Reply:
I think something like this should work:

```
meta.exec()
  .with_context(|| format!("running `cargo metadata --manifest-path {}`", cargo_toml.display()))
```

Reply:
done